egs/ifnenit/v1/run.sh

```diff
@@ -0,0 +1,145 @@
+  local/prepare_dict.sh
+  utils/prepare_lang.sh --num-sil-states 3 --num-nonsil-states 4 --position-dependent-phones false data/local/dict "<unk>" data/local/lang data/lang
+fi
+
+if [ $stage -le 2 ]; then
+  # LM preparation
+  echo "LM preparation"
+  cat data/train/text | cut -d' ' -f2- | utils/make_unigram_grammar.pl | \
+    fstcompile --isymbols=$data_dir/lang/words.txt --osymbols=$data_dir/lang/words.txt > $data_dir/lang/G.fst
+fi
+
+if [ $stage -le 3 ]; then
+  steps/train_mono.sh --nj $nj \
+    $data_dir/train \
+    $data_dir/lang \
+    $exp_dir/mono
+fi
+
+if [ $stage -le 4 ]; then
+  utils/mkgraph.sh --mono $data_dir/lang \
+    $exp_dir/mono \
+    $exp_dir/mono/graph
+  steps/decode.sh --nj $nj --cmd $cmd \
+    $exp_dir/mono/graph \
+    $data_dir/test \
+    $exp_dir/mono/decode_test
+fi
+
+if [ $stage -le 5 ]; then
+  steps/align_si.sh --nj $nj \
+    $data_dir/train $data_dir/lang \
+    $exp_dir/mono \
+    $exp_dir/mono_ali
+  steps/train_deltas.sh \
+    500 20000 $data_dir/train $data_dir/lang \
+    $exp_dir/mono_ali \
+    $exp_dir/tri
+fi
+
+if [ $stage -le 6 ]; then
+  utils/mkgraph.sh $data_dir/lang \
+    $exp_dir/tri \
+    $exp_dir/tri/graph
+  steps/decode.sh --nj $nj --cmd $cmd \
+    $exp_dir/tri/graph \
+    $data_dir/test \
+    $exp_dir/tri/decode_test
+fi
+
+if [ $stage -le 7 ]; then
+  steps/align_si.sh --nj $nj --cmd $cmd \
+    $data_dir/train $data_dir/lang \
+    $exp_dir/mono \
+    $exp_dir/mono_ali
+  steps/train_lda_mllt.sh --cmd $cmd \
+    --splice-opts "--left-context=3 --right-context=3" \
+    500 20000 \
+    $data_dir/train $data_dir/lang \
+    $exp_dir/mono_ali $exp_dir/tri2
+fi
+
+if [ $stage -le 8 ]; then
+  utils/mkgraph.sh $data_dir/lang \
+    $exp_dir/tri2 \
+    $exp_dir/tri2/graph
+  steps/decode.sh --nj $nj --cmd $cmd \
+    $exp_dir/tri2/graph \
+    $data_dir/test \
+    $exp_dir/tri2/decode_test
+fi
+
+if [ $stage -le 9 ]; then
+  steps/align_fmllr.sh --nj $nj --cmd $cmd \
+    --use-graphs true \
+    $data_dir/train $data_dir/lang \
+    $exp_dir/tri2 \
+    $exp_dir/tri2_ali
+  steps/train_sat.sh --cmd $cmd \
+    500 20000 \
+    $data_dir/train $data_dir/lang \
+    $exp_dir/tri2_ali $exp_dir/tri3
+fi
+
+if [ $stage -le 10 ]; then
+  utils/mkgraph.sh $data_dir/lang \
+    $exp_dir/tri3 \
+    $exp_dir/tri3/graph
+  steps/decode_fmllr.sh --nj $nj --cmd $cmd \
+    $exp_dir/tri3/graph \
+    $data_dir/test \
+    $exp_dir/tri3/decode_test
+fi
+
+if [ $stage -le 11 ]; then
+  steps/align_fmllr.sh --nj $nj --cmd $cmd \
+    --use-graphs true \
+    $data_dir/train $data_dir/lang \
+    $exp_dir/tri3 \
+    $exp_dir/tri3_ali
+fi
+
+affix=1a
+affix_chain=1b_chainali
+nnet3_affix=fsf4
+
+if [ $stage -le 12 ]; then
+  local/chain/run_cnn_1a.sh --stage 0 \
+   --gmm tri3 \
+   --ali tri3_ali \
+   --nnet3_affix $nnet3_affix \
+   --affix $affix \
+   --lang_test lang
+fi
+
+if [ $stage -le 13 ]; then
+  local/chain/run_cnn_chainali_1b.sh --stage 0 \
```

Comment:
Make these options default and remove them from here (also for stage 12). See run.sh in iam or uw3 and try to follow their style. Remove all those affix variables as well.

Reply:
I have followed their style in my scripts, but I think there is some problem in iam/v1/local/run_cnn_chainali_1b.sh options. (It won't work with default options).
If you check the options you can realize that the {affix} option which is used in line 30 should not be the same {affix} which is used in line 71. The first one is for the alignment model but the second is for output model.
I have changed my code a little to handle this issue.

Reply:
Yes I've got your point, I did that in my last commit. I only wanted to
mention that problem.